src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        // Формат дат
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+
+        Map<String, Integer> salaryMap = new HashMap<>();
+        for (String name : names) {
+            salaryMap.put(name, 0);
+        }
+
+        for (String entry : data) {
+            String[] parts = entry.split(" ");
+            LocalDate workDate = LocalDate.parse(parts[0], formatter);
+            String name = parts[1];
+            int hoursWorked = Integer.parseInt(parts[2]);
+            int hourlyRate = Integer.parseInt(parts[3]);
+
+            if (!workDate.isBefore(startDate)
+                    && !workDate.isAfter(endDate)
+                    && salaryMap.containsKey(name)) {
```

Comment:
The condition checks if the work date is within the specified range and if the name exists in the `salaryMap`. This is correct, but ensure that the `data` array only contains valid names that are present in the `names` array to avoid unnecessary checks.